network-libp2p/src/discovery/handler.rs

```diff
@@ -169,21 +165,23 @@ pub struct Handler {
 }
 
 impl Handler {
+    const STATE_TRANSITION_TIMEOUT_MS: u64 = 750;
```

Comment:
This is quite a short timeout, some packet loss might already reach that. Not sure what would be a good value.

```suggestion
    const STATE_TRANSITION_TIMEOUT: Duration = Duration::from_millis(5000);
```